demo/vl-map-layer-switcher.html

```diff
@@ -0,0 +1,69 @@
+<!DOCTYPE html>
+<html>
+
+<head>
+    <title>vl-map-layer-switcher voorbeeld</title>
+    <script type="module" src="/node_modules/vl-ui-demo/dist/vl-demo-all.js"></script>
+    <script type="module" src="/node_modules/vl-ui-body/dist/vl-body.js"></script>
+    <script type="module" src="/node_modules/vl-ui-grid/dist/vl-grid.js"></script>
+    <script type="module" src="/node_modules/vl-ui-form-message/dist/vl-form-message.js"></script>
+    <script type="module" src="/node_modules/vl-ui-checkbox/dist/vl-checkbox.js"></script>
+
+    <script type="module" src="/src/vl-map-all.js"></script>
+
+    <link rel="stylesheet" type="text/css" href="/node_modules/vl-ui-body/dist/style.css" />
+    <link rel="stylesheet" type="text/css" href="/node_modules/vl-ui-grid/dist/style.css" />
+    <link rel="stylesheet" type="text/css" href="/node_modules/vl-ui-form-message/dist/style.css" />
+
+    <style>
+        vl-map-search {
+            --vl-search-width: 100%;
+        }
+    </style>
+</head>
+
+<body is="vl-body">
+    <vl-demo-page data-vl-webcomponent="vl-map-layer-switcher">
+        <vl-demo data-vl-title="Kaart met optie om kaartlagen te tonen en verbergen">
+            <div is="vl-grid" data-vl-is-stacked>
+                <div is="vl-column" data-vl-size="12" data-vl-medium-size="12">
+                    <div><label is="vl-form-label">Zoek op adres</label></div>
+                    <vl-map-search></vl-map-search>
+                </div>
+                <div is="vl-column" data-vl-size="12" data-vl-medium-size="12">
+                    <vl-map id="map-with-layer-switcher">
+                        <vl-map-side-sheet>
+                            <vl-map-layer-switcher>
+                                <vl-checkbox data-vl-label="Kaartlaag" data-vl-layer="layer"></vl-checkbox>
+                            </vl-map-layer-switcher>
+                        </vl-map-side-sheet>
```

Comment:
de common case lijkt mij dat ge gewoon per laag een checkbox wil, dus ik zou de API veel meer in die richting maken, dan dat ge individueel moet gaan doorgeven voor welke lagen je al dan niet wil.

of met includes="..." en excludes="..." werken. het feit dat je een checkbox maakt per layer waarmee je hem dan aan of af zit is interne keuken van de layer switcher, en hoeft niet onder controle van de afnemer te zitten.

bijkomend, maar iets minder belangrijk, lijkt mij ook dat de afnemer niet zou moeten zeggen dat dat in een sidesheet moet komen ...

Reply:
dus mij lijkt de API eerder iets als
`
                           <vl-map-layer-switcher includes="*" excludes="....">
                            </vl-map-layer-switcher>
`
te moeten zijn

Reply:
Door de afnemers zelf verantwoordelijk te maken voor het renderen van de input elementen bereiken we de meeste vrijheid. Afnemers kunnen checkboxen of radio input elementen kiezen afhankelijk van de functionele requirements. Bovendien hebben afnemers al de vrijheid om interacties toe te voegen op (click) events en zijn ze zelf verantwoordelijk voor (i18n) label rendering. Wanneer we dit alles afschermen in een layer switcher met includes en excludes beperken we de vrijheid en mogelijkheden te veel of moeten we te veel API functies voorzien om dergelijke zaken te bereiken.

Net zoals de zoek op adres component, kan de kaartlagen switcher op een andere plaatst gerenderd worden. Hierdoor is de side-sheet geen onderdeel van de kaartlagen switcher.

Reply:
Met radio's gaat dit alleszins niet werken, want je zet "onclick" alleen maar die bijhorende laag al dan niet aan of af, en niet diegene die dan door een radio-click zou "af" gaan. Bovendien is een layer switcher voor mij iets om "elke laag individueel aan of uit te zetten", dus ik zie geen use case voor radio's.

Ik zie door naar de demo-pagina te kijken ook inconsistentie dan tussen hoe de overview-map werkt en hoe de layer-switcher werkt. De overview-map gaat zelf op zoek naar alle base layers, de layer switcher zou dat dan niet doen ...

Voor mij geldt altijd bij API design, "make the simple things easy, make the complex things possible", en dat zie ik hier alvast niet terug komen. 
Ik denk dat we dezelfde discussie gevoerd hebben bij vl-wizard, en ik meen toch opgevangen te hebben bij andere teams dat daar toch wat struggle geweest is, om hem in gebruik te nemen (al weet ik er niet genoeg van om te beweren of dat al dan niet aan het API design lag).
Ik ben absoluut akkoord dat je met deze API ALLES kunt maar ik mis dus het stukje waar ik iets eenvoudigs MAKKELIJK kan.

Anders moeten we via teams hier nog eens over sync'en, bel me gerust eens op morgen ...

Reply:
ook met @SanderKleykens nog eens overlegd, misschien moet je dan ook een vl-simple-layer-switcher of zo maken, die onze noden dekt?

Reply:
Dat alles mogelijk is, lijkt me een al goed startpunt.

Verbeteringen om het voor afnemers nog eenvoudiger te maken is mogelijk. We moeten alleen rekening houden dat er verschillende soorten afnemers zijn en het vaak moeilijk is om een default te bepalen die voor iedereen goed is. Bij de vl-ui-wizard was het bepalen van deze defaults één van de problemen. Daar werd uiteindelijk een API uitbreiding voorzien via een attribuut zodat eenvoudig bereikt kon worden wat voor één afnemer de default moest zijn maar door UIG niet als standaard geïmplementeerd werd.

In dit geval lijkt het mij een eventuele zinvolle uitbreiding om een default te voorzien die voor elke kaartlaag een checkbox toevoegt waardoor op elk moment de visibiliteit regelbaar is. De Voortoets kan deze default helaas niet gebruiken want er zijn kaartlagen die maar zichtbaar worden vanaf een bepaald zoom niveau o.a. GRB-Percelenkaart, GRB-Gebouwenlaag, Speciale Beschermings Zones, Biologische Waarderingskaart en Voorlopige zoekzones. De default moet dan al snel uitgebreid worden met resolutions. Het lijkt me ook redelijk standaard dat de visibiliteit van bepaalde kaartlagen niet getoond mogen worden, war dan weer een includes en excludes uitbreiding vraagt.

Het grote verschil met de vl-ui-overview-map is dat er geen basis kaartlagen zijn die niet aangeduid kunnen worden door de afnemer want dan zouden deze niet aanwezig zijn.

We zullen morgen even syncen.

Reply:
Na overleg werd besloten om een default implementatie te voorzien zoals hierboven beschreven. Voorlopig dekt dit de lading, maar kan achteraf nog verder verfijnd worden indien nodig.

Reply:
(y)

Reply:
👍